test/state/precompiles_silkpre.cpp

```diff
@@ -0,0 +1,76 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2022 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "precompiles_silkpre.hpp"
+#include <silkpre/precompile.h>
+#include <cassert>
+#include <cstring>
+
+namespace evmone::state
+{
+namespace
+{
+ExecutionResult execute(const uint8_t* input, size_t input_size, uint8_t* output_buf,
+    [[maybe_unused]] size_t max_output_size, PrecompileId id) noexcept
+{
+    const auto index = static_cast<size_t>(id) - 1;
+    const auto [output, output_size] = kSilkpreContracts[index].run(input, input_size);
+    if (output == nullptr)
+        return {EVMC_PRECOMPILE_FAILURE, 0};
+    assert(output_size <= max_output_size);
```

Comment:
Why is this an assertion? should you copy `max(max_output_size, output_size)` bytes?

Reply:
The `max_output_size` is computed by evmone's analysis and it should match what silkpre computes. But using `max` sounds good while keeping the assert.